Support a `danger local` command

orta opened the pull request:
Re: https://github.com/KrauseFx/danger/issues/17

So far I've added another commands `danger local` and set it up to use a non-existent  CI Source, the source's job is going to be to find the last PR it can find in the current history of the current repo, then fill in it's environment variables with that stuff.

orta commented:
Now grabs enough git data :tada:

KrauseFx commented on lib/danger/ci_source/local_git_repo.rb:
```diff
@@ -0,0 +1,43 @@
+          self.pull_request_id = pr_merge.replace("Merge pull request #", "").to_i
+          self.base_commit = pr_merge.parents[0].sha
+          self.head_commit = pr_merge.parents[1].sha
+        end
```
What happens on `else`?

orta replied:
I'll do a full pass on error reporting once I've got a working concept :+1:

KrauseFx replied:
:+1:

KrauseFx commented:
Looks good, however HoundCI doesn't seem so happy

orta commented:
also, if you have a better idea for the name, I'd be interested

orta commented on lib/danger/dangerfile.rb:
```diff
@@ -21,9 +21,32 @@ def to_s
+      puts "           #{'HEAD commit'.blue} : #{env.ci_source.head_commit}"
+      puts "           git diff  #{env.ci_source.base_commit} #{env.ci_source.head_commit}".yellow
+      puts "#{'Requests'.blue} : #{env.request_source.class}"
+    end
```
this should be replaced with cork, really

orta replied:
It's a start
![screen shot 2016-02-08 at 10 43 56](https://cloud.githubusercontent.com/assets/49038/12883599/deb2b90e-ce50-11e5-8e49-2181a642e1c6.png)

segiddins replied:
terminal-table may be what you want here

KrauseFx replied:
Yeah, so simple to use and nice looking https://github.com/fastlane/fastlane_core/blob/master/lib/fastlane_core/print_table.rb#L19-L25

orta replied:
Cool, I was wondering if Cork would do this? Can get another external user in this case

orta replied:
but yeah, terminal-table looks great

orta commented on lib/danger/commands/local.rb:
```diff
@@ -0,0 +1,45 @@
+    end
+
+    def run
+      ENV["DANGER_USE_LOCAL_GIT"] = "YES"
```
all `ci_source`s use ENV for validation, I think using it here makes sense.

orta commented on lib/danger.rb:
```diff
@@ -1,8 +1,7 @@
 require "danger/environment_manager"
-require "danger/runner"
-require "danger/init"
+require "danger/commands/runner"
```
tried to make this work like cocoapods, but I'm missing something

orta replied:
CocoaPods shows the sub-commands
```
~/d/r/l/danger (local_test) ⏛  pod --help
Usage:

    $ pod COMMAND

      CocoaPods, the Cocoa library package manager.

Commands:

    + cache          Manipulate the CocoaPods cache
    + deintegrate    Deintegrate CocoaPods from your project.
    + dependencies   Show project's dependency graph.
```

Danger doesn't
```
~/d/r/l/danger (local_test) ⏛  bundle exec danger --help
Usage:

    $ danger COMMAND

      Run the Dangerfile.

Options:

    --version   Show the version of the tool
    --verbose   Show more debugging information
```

orta commented on lib/danger/commands/runner.rb:
```diff
@@ -18,6 +21,7 @@ def validate!
     def run
       # The order of the following commands is *really* important
       dm = Dangerfile.new
+      dm.verbose = verbose
```
every claide object has a verbose

orta commented:
OK, I've removed match, and fui - it wasn't being used. 

I've also updated Danger to this PR https://github.com/KrauseFx/danger/pull/40/ - which makes `--verbose` work - should make debugging DM easier for me.

orta commented:
Punted the tables stuff to https://github.com/danger/danger/issues/62
